### PR TITLE
fix IsOptsWithFromKey

### PR DESCRIPTION
**Pull request by LeoYang90**
fix https://github.com/etcd-io/etcd/issues/13332

### CODE REVIEW

File: client/v3/op.go [hunk 1]
```diff
@@ -554,7 +559,21 @@ func toLeaseTimeToLiveRequest(id LeaseID, opts ...LeaseOption) *pb.LeaseTimeToLi
 }
 
 // IsOptsWithPrefix returns true if WithPrefix option is called in the given opts.
-func IsOptsWithPrefix(opts []OpOption) bool { return isOpFuncCalled("WithPrefix", opts) }
```
Comment (serathius) [inline client/v3/op.go:557]: Looks like isOpFuncCalled is no longer used. Can you remove it?

**Review by serathius — COMMENTED** [review]
Thanks for looking into this. Aside of one nit this looks good. 

I'm still struggling to understand how this issue got on main branch, and why I'm unable to reproduce it locally.
Do you have any idea?

**Comment by serathius** [comment]
Also can you squash commits? second commit in the PR just removes trailing whitespace. It would be better if this is put in single commit.

**Review by serathius — APPROVED** [review]
Would be inclined to merge it as it is to unblock other PRs.

File: tests/integration/testing_test.go [hunk 2]
```diff
@@ -24,5 +24,5 @@ import (
 func TestBeforeTestWithoutLeakDetection(t *testing.T) {
 	integration.BeforeTest(t, integration.WithoutGoLeakDetection(), integration.WithoutSkipInShort())
 	// Intentional leak that should get ignored
-	go time.Sleep(2 * time.Second)
+	go time.Sleep(200 * time.Millisecond)
```
Comment (ptabor) [inline tests/integration/testing_test.go:27]: +1. Indeed it was irresponsible. Thank you for catching.

File: client/v3/op.go [hunk 3]
```diff
@@ -554,7 +559,21 @@ func toLeaseTimeToLiveRequest(id LeaseID, opts ...LeaseOption) *pb.LeaseTimeToLi
...
+func IsOptsWithPrefix(opts []OpOption) bool {
+	ret := &Op{key: []byte("")}
```
Comment (ptabor) [inline client/v3/op.go:563]: It would be cleaner to have `newOp()` factory method that takes care of setting the default versions of the fields/options.

**Review by ptabor — APPROVED** [review]
Thank you for fixing.

**Comment by tsaarni** [comment]
> I'm still struggling to understand how this issue got on main branch, and why I'm unable to reproduce it locally.
> Do you have any idea?

Go version got upgraded in github action https://github.com/etcd-io/etcd/pull/13311#issuecomment-915770306

**Comment by serathius** [comment]
@tsaarni Nice find, but also terrifying. I think we should fix it to make test reproducible.

**Comment by ptabor** [comment]
All green. Great.  Merging.

**Comment by vivekpatani** [comment]
Hi folks, any plans to port this back to `release-3.5`? @serathius @ptabor @LeoYang90
If it's not already done, I can help port it back. Thanks.

**Comment by ptabor** [comment]
Thank you, Vivek. Porting seems right to me. Will approve the PR.

**Pull request merged**